src/compiler/scala/tools/nsc/typechecker/TypeDiagnostics.scala

```diff
@@ -584,13 +584,14 @@ trait TypeDiagnostics {
 
         def unusedTypes = defnTrees.toList.filter(t => isUnusedType(t.symbol)).sortBy(treepos)
         def unusedTerms = {
+          def accessSameField(getter: Symbol, setter: Symbol): Boolean =
+            getter.accessed == setter.accessed
           val all = defnTrees.toList.filter(v => isUnusedTerm(v.symbol))
 
-          // filter out setters if already warning for getter, indicated by position.
-          // also documentary names in patterns.
+          // filter out setters if already warning for getter
           all.filterNot(v =>
-              v.symbol.isSetter && all.exists(g => g.symbol.isGetter && g.symbol.pos.point == v.symbol.pos.point)
-           || atBounds.exists(x => v.symbol.pos.point == x.pos.point)
+              v.symbol.isSetter && all.exists(g => g.symbol.isGetter && accessSameField(g.symbol, v.symbol))
+           || atBounds.exists(x => x.isGetter && accessSameField(v.symbol,  x))
```

Comment:
The or is for "documentary names", see the deleted comment, by which I meant `case X(i @ _) =>` where this form disables the unused check for `i`. Someone I suggest that named args in patterns might serve the same purpose.